### PR TITLE
Added Hidden Field Type

### LINKED ISSUE #370: Add hidden field type

### PR DESCRIPTION
**Pull request by mikeselander**
Added a `hidden` field type since this was the #1 desired field type by HM staff.

Resolves #370

*I have:*
 - [x] Run WordPress VIP PHPCS check and code parses without errors
 - [x] Added any new PHPUnit tests
 - [x] Run PHPUnit and all tests are passing after adding any necessary new tests
 - [x] Tested feature/bugfix on single and multisite install

### CODE REVIEW

File: custom-meta-boxes.php [hunk 1]
```diff
@@ -39,6 +39,7 @@
  */
 include_once( CMB_PATH . '/classes.fields.php' );
 include_once( CMB_PATH . '/class.cmb-meta-box.php' );
+include_once( CMB_PATH . 'fields/class-cmb-hidden-field.php' );
```
Comment (rmccue) [inline custom-meta-boxes.php]: Slash here?

File: fields/class-cmb-hidden-field.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		?>
+
+		<div class="field-item" data-class="<?php echo esc_attr( get_class( $this ) ); ?>" style="display: none; <?php echo esc_attr( $this->args['style'] ); ?>">
+
+			<?php $this->html(); ?>
+
+		</div>
+
+		<?php
+	}
```
Comment (rmccue) [inline fields/class-cmb-hidden-field.php]: Is there any point having a `<div>` at all?

Comment (mikeselander) [inline fields/class-cmb-hidden-field.php]: Yes, I'd like to keep the markup consistent and it's used for JS checks so if we needed to make any transformations to the field in the future it keeps it consistent.

Comment (rmccue) [inline fields/class-cmb-hidden-field.php]: Radical.

**Pull request merged**